consensus/istanbul/backend/engine.go

```diff
@@ -390,6 +390,26 @@ func (sb *Backend) Prepare(chain consensus.ChainReader, header *types.Header) er
 	// wait for the timestamp of header, use this to adjust the block period
 	delay := time.Unix(header.Time.Int64(), 0).Sub(now())
 	time.Sleep(delay)
+
+	// modify the block header to include all the ParentCommits
+	parentCommittedSeals := sb.core.ParentCommits()
+	if parentCommittedSeals != nil {
+		if parentCommittedSeals.Size() != 0 {
+			sb.logger.Info("Storing", "parent seals from", parentCommittedSeals.String())
+			bitmap, asig := istanbulCore.AggregateSeals(parentCommittedSeals)
+			writeCommittedSeals(header, bitmap, asig, true)
+		}
+	} else {
+		// if we were not gossipped any commits, then just copy over the seals
```

Comment:
We should also be wary of the case when the parent commit messages are for a different block than what winds up in our chain. In *theory* I don't think this should happen (right @kevjue?) but I'd like to check the digests just in case

Reply:
Ya, that shouldn't happen.  All blocks that are inserted in the chain have their header verified: https://github.com/celo-org/celo-blockchain/blob/master/core/blockchain.go#L1137